addons/ui/functions/fnc_setAdvancedElement.sqf

```diff
@@ -18,21 +18,29 @@
  * Public: No
  */
 
-params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]] ];
+params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]]];
 
 private _cachedElement = GVAR(configCache) getVariable _element;
-if (isNil "_cachedElement") exitWith {};
+if (isNil "_cachedElement") exitWith {TRACE_1("nil element",_this)};
 
 if (!_force && {!GVAR(allowSelectiveUI)}) exitWith {
+    TRACE_1("not allowed",_this);
     [LSTRING(Disallowed), 2] call EFUNC(common,displayTextStructured);
     false
 };
 
 _cachedElement params ["_idd", "_elements", "_location", "_conditions"];
 
 // Exit if main vehicle type condition not fitting
-private _canUseWeapon = ACE_player call CBA_fnc_canUseWeapon;
-if ((_canUseWeapon && {_location == 2}) || {!_canUseWeapon && {_location == 1}}) exitWith {false};
+private _cargoIndex = vehicle ACE_player getCargoIndex ACE_player; // nil if not in vehicle
```

Comment:
Why check cargo specifically? Instead of just in vehicle check?

Reply:
because these controls (Soldier/Vehicle/Weapon Information, `location = GROUND_ONLY`) should be set when on foot or in vehicle cargo including FFV.

Reply:
Yeah but why cargo? Why exclude driver/gunner?

Reply:
@jonpas should know. I think because that info is shown while player is in cargo and isn't shown when in other seats.

Reply:
Why exactly does old way break with moveInCargo?

Reply:
I see, I will test this soon.